laws/shared/src/main/scala/algebra/laws/LogicLaws.scala

```diff
@@ -56,9 +56,26 @@ trait LogicLaws[A] extends LatticeLaws[A] {
     "(0 → x) = 1" -> forAll { (x: A) => A.imp(A.zero, x) ?== A.one }
   )
 
+  def generalizedBool(implicit A: GenBool[A]) = new LogicProperties(
+    name = "generalized bool",
+    parents = Seq(),
+    ll = new LatticeProperties(
+      name = "lowerBoundedDistributiveLattice",
+      parents = Seq(boundedJoinSemilattice, distributiveLattice),
+      join = Some(boundedSemilattice(A.joinSemilattice)),
+      meet = Some(semilattice(A.meetSemilattice))
+    ),
+
+    "x∖y ∧ y = 0" -> forAll { (x: A, y: A) =>
```

Comment:
Hmm, there is a Unicode set difference symbol instead of a regular backslash used here because it is inside a string literal. Turns out they are not rendered correctly :/